zulipterminal/ui_tools/views.py

```diff
@@ -901,9 +901,12 @@ def __init__(self, controller: Any, title: str, *, zt_version: str,
 
 class MsgLinksView(PopUpView):
     def __init__(self, controller: Any, msg: Message, title: str) -> None:
-        width = len(title) + 4
+        width = max(len(title) + 4, 64)
         widgets = []
-
+        for index, (text, link) in enumerate(msg['links'].items(), start=1):
+            widgets.append(
+                urwid.Text('[{}]: {} - {}'.format(index, text, link))
```

Comment:
As link can be extremely long, I am inclining towards only showing text with mark - `[1]: Text` in `MsgLinksView`. Thoughts?

Reply:
Also, I haven't used `PopUpView` here as I am planning to use buttons from Shreya Malviya's old PR later (for handling internal/external links).

Reply:
I'm not sure precisely what you mean here, but perhaps this warrants discussion in light of checkboxes being used with @kaustubh-nair 's stream settings work, to work towards a general solution?

Reply:
Since both PRs are being worked on simultaneously, I think it would be better if we merge them both first and then do a refactor - this would save time as well as make it easier to ensure that the refactor works for both the PRs

Reply:
In the latest update, I have used the existing code from `PopUpView` but, yes, the use-case will likely evolve if we decide to incorporate new changes over the course of time. I believe it would be reasonable to refactor later.